compiler/luci/pass/src/FuseBCQPass.cpp

```diff
@@ -20,26 +20,341 @@
+        }
+      }
+    }
+
+    return packed_binary;
+  }
+
+  /**
+   * @brief Exclude BCQ information nodes which are used for fusing BCQ operations
+   *        from graph output by using CircleOutputExclude
+   */
+  void clear_BCQ_nodes()
+  {
+    auto createNoOp = [](luci::CircleConst *const_node) {
+      auto graph = const_node->graph();
+      auto noOp = graph->nodes()->create<luci::CircleOutputExclude>();
+
+      noOp->dtype(const_node->dtype());
+      noOp->rank(const_node->rank());
+      for (uint32_t i = 0; i < const_node->rank(); ++i)
+        noOp->dim(i) = const_node->dim(i);
+
+      return noOp;
+    };
+
+    for (auto &n : const_nodes_alphas_const)
+    {
+      auto node = n.second;
+
+      for (auto s : loco::succs(node))
+      {
+        if (auto outnode = dynamic_cast<luci::CircleOutput *>(s))
+        {
+          outnode->from(createNoOp(node));
+        }
+      }
+    }
+
+    for (auto &n : const_nodes_num_quantized_bits_const)
+    {
+      auto node = n.second;
+      loco::replace(node).with(createNoOp(node));
+    }
+
+    for (auto &n : const_nodes_bits_const)
+    {
+      auto node = n.second;
+      loco::replace(node).with(createNoOp(node));
+    }
+
+    for (auto &n : const_nodes_per_col_alphas_const)
+    {
+      auto node = n.second;
+      loco::replace(node).with(createNoOp(node));
+    }
+  }
+
+  bool is_valid_BCQ(luci::CircleConst *node)
+  {
+    auto alphas = get_alphas(node);
+    auto binary = get_binary(node);
+
+    if (alphas->dim(0).value() != binary->dim(1).value())
+      return false;
+
+    if (alphas->dim(1).value() != binary->dim(0).value())
+      return false;
+
+    // TODO Support when hidden size cannot be divided by 32
+    if (binary->dim(2).value() % 32 != 0)
+      return false;
+
+    int bits_size = alphas->dim(1).value();
+    int out_size = alphas->dim(0).value();
+    int hidden_size = binary->dim(2).value();
+
+    // Quantized value validation
+    for (int o = 0; o < out_size; ++o)
+    {
+      for (int h = 0; h < hidden_size; ++h)
+      {
+        float res = 0.0;
+        for (int b = 0; b < bits_size; ++b)
+        {
+          if (binary->at<loco::DataType::BOOL>(b * out_size * hidden_size + o * hidden_size + h) ==
+              1)
+            res += alphas->at<loco::DataType::FLOAT32>(o * bits_size + b);
+          else
+            res -= alphas->at<loco::DataType::FLOAT32>(o * bits_size + b);
+        }
+
+        auto diff = res - node->at<loco::DataType::FLOAT32>(o * hidden_size + h);
+        if (diff < -0.0001 || 0.0001 < diff)
+          return false;
+      }
+    }
+
+    return true;
+  }
+
+private:
+  std::map<std::string, luci::CircleConst *> const_nodes_alphas_const;
+  std::map<std::string, luci::CircleConst *> const_nodes_bits_const;
+  std::map<std::string, luci::CircleConst *> const_nodes_num_quantized_bits_const;
+  std::map<std::string, luci::CircleConst *> const_nodes_per_col_alphas_const;
```

Comment:
(optional) it would be better to add `_` in front of member variables like others